packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/lambda/call-aws-service-cross-region.ts

```diff
@@ -0,0 +1,182 @@
+ */
+export class CallAwsServiceCrossRegion extends sfn.TaskStateBase {
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+  protected readonly lambdaFunction: IFunction;
+
+  constructor(scope: Construct, id: string, private readonly props: CallAwsServiceCrossRegionProps) {
+    super(scope, id, props);
+
+    if (props.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
+      throw new Error('The RUN_JOB integration pattern is not supported for CallAwsService');
+    }
+    if (!Token.isUnresolved(props.action) && !props.action.startsWith(props.action[0]?.toLowerCase())) {
+      throw new Error(`action must be camelCase, got: ${props.action}`);
+    }
+    if (props.parameters) {
+      const invalidKeys = Object.keys(props.parameters).filter((key) => !key.startsWith(key[0]?.toUpperCase()));
+      if (invalidKeys.length) {
+        throw new Error(`parameter names must be PascalCase, got: ${invalidKeys.join(', ')}`);
+      }
+    }
+
+    // props.service expects a service name in the AWS SDK for JavaScript v3 format.
+    // In some services, this format differs from the one used in IAM.
+    // We try to automatically convert those formats here (not exhaustive though).
+    const iamServiceMap: Record<string, string> = {
+      cloudwatchlogs: 'cloudwatch-logs',
+      mediapackagevod: 'mediapackage-vod',
+    };
+    const iamService = iamServiceMap[props.service] ?? props.service;
+
+    this.lambdaFunction = new CrossRegionAwsSdkSingletonFunction(this, 'Handler', {
+      uuid: '8a0c93f3-dbef-4b71-ac13-7aaf2048ce7e',
+      lambdaPurpose: 'CrossRegionAwsSdk',
+      timeout: Duration.seconds(30),
+    });
+
+    [
+      new iam.PolicyStatement({
+        resources: props.iamResources,
+        // The prefix and the action name are case insensitive
+        // https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_action.html
+        actions: [props.iamAction ?? `${iamService}:${props.action}`],
+      }),
+      ...(props.additionalIamStatements ?? []),
+    ].forEach((policy) => this.lambdaFunction.addToRolePolicy(policy));
+
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: this.lambdaFunction.resourceArnsForGrantInvoke,
+        actions: ['lambda:InvokeFunction'],
+      }),
+    ];
+
+    if (props.retryOnServiceExceptions ?? true) {
+      // Best practice from https://docs.aws.amazon.com/step-functions/latest/dg/bp-lambda-serviceexception.html
+      this.addRetry({
+        errors: ['Lambda.ClientExecutionTimeoutException', 'Lambda.ServiceException', 'Lambda.AWSLambdaException', 'Lambda.SdkClientException'],
+        interval: Duration.seconds(2),
+        maxAttempts: 6,
+        backoffRate: 2,
+      });
+    }
+  }
+
+  /**
+   * Provides the Lambda Invoke service integration task configuration
+   */
+  /**
+   * @internal
+   */
```

Comment:
Any reason to split the TSDoc?
```suggestion
  /**
   * Provides the Lambda Invoke service integration task configuration
   *
   * @internal
   */
```